md: Moved argument capture into MDInstance init

dlsloan opened the pull request:
allocating special devices requires knowlege of the devices to be used
for testing and test3 attempts to allocate a verification device before
setting up the md instance in order to zero the devices before starting
the test. Fix this by getting arguments and storing them in MDInstance
init for later use by get_special_device() and setup().

While we're at it correct a small bug in the verify md_dev lookup

dlsloan committed:
md: Moved argument capture into MDInstance init

allocating special devices requires knowlege of the devices to be used
for testing and test3 attempts to allocate a verification device before
setting up the md instance in order to zero the devices before starting
the test. Fix this by getting arguments and storing them in MDInstance
init for later use by get_special_device() and setup().

dlsloan committed:
test3: test3 runner had incorrect md_dev lookup attribute

Add self._md.md_dev to correctly access md_dev

lsgunth commented on test3:
```diff
@@ -102,7 +102,7 @@ class TestRunner:
             print("Verifying")
             subprocess.check_call(["cmp", "-b", "-n", str(VERIFICATION_SZ),
-                                   self.md_dev, self.verify_disk])
+                                   self._md._md_dev, self.verify_disk])
```
Probably need to remove the single underscore from this if we are going to use it outside of its class....

Otherwise, this looks fine to me. I have a couple patches that are going to conflict though.

dlsloan replied:
Which direction is going to be easier to resolve?

lsgunth replied:
Doing yours first

dlsloan committed:
md.py: Make md_dev public by dropping '_'

md_dev is used by test3, drop private naming for the test3 fix in the
next patch.

lsgunth reviewed (approved):
LGTM

Minor nit: the FIXUP tag is weird an non-standard to see in a pull request.

iomartin reviewed (approved):
Looks good to me

dlsloan commented:
> Minor nit: the FIXUP tag is weird an non-standard to see in a pull request.

Dropped the FIXUP tag in the commit message
